### ISSUE TITLE
Failure to Log with Custom Trainer

### ISSUE THREAD

**Issue by twesholmes** [MIC-DKFZ/nnUNet#2959]
Hi,

I have created a custom trainer. When I run nnUNetv2_train with the -tr flag, the network does train but I get an error of "failed to log". The custom trainer appears to be behaving as expected as it stops after 250 epochs, but the error persists throughout each epoch several times.

My custom trainer is:
```
import torch
from nnunetv2.training.nnUNetTrainer.nnUNetTrainer import nnUNetTrainer

class MyCustomTrainer_V01(nnUNetTrainer):
    def __init__(self, plans: dict, configuration: str, fold: int, dataset_json: dict, 
                 device: torch.device = torch.device('cuda')):
        super().__init__(plans, configuration, fold, dataset_json, device)
        self.num_epochs = 250
```

The full error message is:
failed to log:  (<class 'FileNotFoundError'>, FileNotFoundError(2, 'No such file or directory'), <traceback object at 0x000001D4DDE15D80>)

The traceback object does change but in groups of 5, after which an expected print statement is made (Epoch 0, Current Learning Rate: 0.01, etc.)

I was able to turn off these error log notifications by commenting out these lines of the print_to_log_file function within nnUNetTrainer.py:
```
except IOError:
    # print(f"{datetime.fromtimestamp(timestamp)}: failed to log: ", sys.exc_info())
    # sleep(0.5)
    ctr += 1
```
But I know that this is not the correct way to handle this issue, and I would like to keep/see the logs.

I am calling the custom trainer by using this command:
nnUNetv2_train 53 2d 0 -tr MyCustomTrainer_V01

Any suggestions on what to change?

Thank you for your help and for creating such an amazing tool!

**Comment by Kobalt93** [comment]
Hi,
- Is the system variable `nnUNet_results` set correctly?
- Does the output folder for this run exist?
- Do you have write permissions to that folder?
- Are you running locally or on a compute cluster?

**Comment by Karol-G** [comment]
Closing this issue due to inactivity.
